### PR TITLE
ensure UIAccess before sending SAS

### PR DESCRIPTION
**Pull request by seanbudd**
According to [win32 API docs](https://docs.microsoft.com/en-us/windows/win32/api/sas/nf-sas-sendsas) `SendSAS` requires UI Access. 

>To successfully call the SendSAS function, an application must either be running as a service or have the uiAccess attribute of the requestedExecutionLevel element set to "true" in its application manifest. If an application is not running as a service, it must be running as either the current user or the LocalSystem account to call SendSAS. In addition, if an application is not running as a service, User Account Control must be turned on to call SendSAS.

>Important  Applications with the uiAccess attribute set to "true" must be signed by using Authenticode. In addition, the application must reside in a protected location in the file system. Currently, there are two allowable protected locations:
\Program Files\\
\windows\system32\

This would mean portable, or otherwise unsigned copies of NVDA would not be able to simulate `CTRL+ALT+DEL`. This PR adds a logged message when that isn't possible based on `systemUtils.hasUiAccess`.

### CODE REVIEW

File: addon/globalPlugins/remoteClient/local_machine.py [hunk 1]
```diff
@@ -77,4 +79,13 @@ def set_clipboard_text(self, text, **kwargs):
...
-		ctypes.windll.sas.SendSAS(0)
+		"""
+		This function simulates as "a secure attention sequence" such as CTRL+ALT+DEL.
+		SendSAS requires UI Access, so we provide a warning when this fails.
+		This warning will only be read by the remote NVDA if it is currently connected to the machine.
+		"""
+		if systemUtils.hasUiAccess():
+			ctypes.windll.sas.SendSAS(0)
+		else:
+			ui.message("No permission on remote device to trigger CTRL+ALT+DEL")
```
Comment (ctoth) [inline addon/globalPlugins/remoteClient/local_machine.py:90]: Please make this message translatable by wrapping the string in _()

File: addon/globalPlugins/remoteClient/local_machine.py [hunk 2]
```diff
@@ -10,6 +10,8 @@
 import ctypes
 import braille
 import inputCore
+import systemUtils
```
Comment (lukaszgo1) [inline addon/globalPlugins/remoteClient/local_machine.py:13]: Currently NVDA Remote claims to be compatible with 2019.3 and later however `systemUtils` has been introduced in 2020.2 in https://github.com/nvaccess/nvda/pull/10493/. The following should ensure backwards compat:
```suggestion

try:
	from systemUtils import hasUiAccess
except ModuleNotFoundError:
	from config import hasUiAccess
	
```

Comment (seanbudd) [reply addon/globalPlugins/remoteClient/local_machine.py:13]: Great catch, thank you

File: addon/globalPlugins/remoteClient/local_machine.py (hunk 1 above)
Comment (seanbudd) [reply addon/globalPlugins/remoteClient/local_machine.py:90]:
```suggestion
			ui.message(_("No permission on remote device to trigger CTRL+ALT+DEL"))
```

File: addon/globalPlugins/remoteClient/local_machine.py [hunk 3]
```diff
@@ -77,4 +84,13 @@ def set_clipboard_text(self, text, **kwargs):
...
-		ctypes.windll.sas.SendSAS(0)
+		"""
+		This function simulates as "a secure attention sequence" such as CTRL+ALT+DEL.
+		SendSAS requires UI Access, so we provide a warning when this fails.
+		This warning will only be read by the remote NVDA if it is currently connected to the machine.
+		"""
+		if systemUtils.hasUiAccess():
+			ctypes.windll.sas.SendSAS(0)
+		else:
+			ui.message(_("No permission on remote device to trigger CTRL+ALT+DEL"))
```
Comment (seanbudd) [inline addon/globalPlugins/remoteClient/local_machine.py:95]:
```suggestion
			# Translators: Sent when a user is fails send CTRL+ALT+DEL from a remote NVDA instance
			ui.message(_("No permission on device to trigger CTRL+ALT+DEL from remote"))
```

File: addon/globalPlugins/remoteClient/local_machine.py [hunk 4]
```diff
@@ -77,4 +84,14 @@ def set_clipboard_text(self, text, **kwargs):
 		api.copyToClip(text=text)
 
 	def send_SAS(self, **kwargs):
-		ctypes.windll.sas.SendSAS(0)
+		"""
+		This function simulates as "a secure attention sequence" such as CTRL+ALT+DEL.
+		SendSAS requires UI Access, so we provide a warning when this fails.
+		This warning will only be read by the remote NVDA if it is currently connected to the machine.
+		"""
+		if systemUtils.hasUiAccess():
```
Comment (lukaszgo1) [inline addon/globalPlugins/remoteClient/local_machine.py:92]: After latest changes this should be:
```suggestion
		if hasUiAccess():
```

File: addon/globalPlugins/remoteClient/local_machine.py [hunk 5]
```diff
@@ -77,4 +84,14 @@ def set_clipboard_text(self, text, **kwargs):
...
-		ctypes.windll.sas.SendSAS(0)
+		"""
+		This function simulates as "a secure attention sequence" such as CTRL+ALT+DEL.
+		SendSAS requires UI Access, so we provide a warning when this fails.
+		This warning will only be read by the remote NVDA if it is currently connected to the machine.
+		"""
+		if systemUtils.hasUiAccess():
+			ctypes.windll.sas.SendSAS(0)
+		else:
+			# Translators: Sent when a user is fails send CTRL+ALT+DEL from a remote NVDA instance
```
Comment (lukaszgo1) [inline addon/globalPlugins/remoteClient/local_machine.py:95]: This message does not make sense.

**Review by lukaszgo1 — CHANGES_REQUESTED** [review]

File: addon/globalPlugins/remoteClient/local_machine.py (hunk 5 above)
Comment (seanbudd) [reply addon/globalPlugins/remoteClient/local_machine.py:95]:
```suggestion
			# Translators: Sent when a user fails to send CTRL+ALT+DEL from a remote NVDA instance
```

**Comment by ctoth** [comment]
Looks like this got a bit out of sync with master. If you can bring it up to date I'd be happy to merge. Thank you very much for your help with this!

**Comment by seanbudd** [comment]
No problem @ctoth - I've merged in the latest master now!

**Comment by ctoth** [comment]
Thanks for your work on this!

**Pull request merged**